trimesh/visual/material.py

```diff
@@ -793,25 +852,92 @@ def material_to_img(mat):
     assert set(np.concatenate(mat_idx).ravel()) == set(range(len(uvs)))
     assert len(uvs) == len(materials)
 
+    use_pbr = any(isinstance(m, PBRMaterial) for m in materials)
+
     # collect the images from the materials
-    images = [material_to_img(materials[g[0]]) for g in mat_idx]
+    images = [get_base_color_texture(materials[g[0]]) for g in mat_idx]
+    unpadded_sizes = [np.array(img.size) for img in images]
+
+    if len(images) <= 1:
```

Comment:
I am not 100% sure about the padding. trimsheet islands are always difficult to handle and the padding can fix interpolation between different islands.
But when I add 2 pixels to a 1024x1024 texture, the output will be a 2048x2048 texture, because we use the next power of two